alf/trainers/policy_trainer.py

```diff
@@ -662,18 +674,38 @@ def play(root_dir,
             buffer_size=num_episodes, reward_shape=env.reward_spec().shape),
         alf.metrics.AverageEpisodeLengthMetric(buffer_size=num_episodes),
     ]
+
+    if future_steps > 0:
+        defer_mode = True
+        observations = []
+        rewards = []
+        actions = []
```

Comment:
Storing all quantities seems memory inefficient if one episode is too long. I think an alternative is to render t-K:t at step t (looking back but pretending to look into the future). In this way, you only need to maintain a window of length K at most and can render it online.

Reply:
It would be great to have this, probably can also simplify code a bit.

Reply:
done